src/x/server/server.go

```diff
@@ -140,16 +146,86 @@ func (s *server) Serve(l net.Listener) error {
 	return nil
 }
 
+func (s *server) upgradeToTLS(conn BufferedConn) (BufferedConn, error) {
+	certPool := x509.NewCertPool()
+	if s.tlsOpts.ClientCAFile() != "" {
+		certs, err := os.ReadFile(s.tlsOpts.ClientCAFile())
```

Comment:
Are we reading this file newly on every connection? That's probably a bad idea (at the least, it seems unnecessary).